File: SOC/src/cc/game/soc/core/BotNodeRoute.java
package cc.game.soc.core;

import cc.lib.game.IVector2D;

public class BotNodeRoute extends BotNode {

	private Route route;
	private int index;
	
	public BotNodeRoute() {}
	
	BotNodeRoute(Route route, int index) {
		this.route = route;
		this.index = index;
	}
	
	@Override
	public Object getData() {
		return index;
	}

	@Override
	public String getDescription() {
		return "E(" + index + ") " + route.toString();
	}

	public IVector2D getBoardPosition(Board b) {
    	return b.getRouteMidpoint(route);
    }

}

File: SOC/src/cc/game/soc/core/BotNodeRoute.kt
package cc.game.soc.core

import cc.lib.game.IVector2D

class BotNodeRoute(val route:Route, val index:Int) : BotNode() {

	override val data = index

	override val description: String = "E($index) $route"

	override fun getBoardPosition(b: Board): IVector2D {
		return b.getRouteMidpoint(route)
	}
}